apps/sms/test/marionette/mocks/mock_navigator_moz_mobile_message.js

```diff
@@ -192,12 +232,15 @@ Services.obs.addObserver(function(document) {
             } else {
               cursor.result = thread.messages[currentIndex];
             }
-            cursor.onsuccess.call(cursor);
+
+            window.setTimeout(function nextTick() {
```

Comment:
So as promised :) Here I tried to use ```Promise.resolve().then(cursor.onsuccess.call(cursor))``` instead, but on ~50 iterations it fails with "too much recursion" exception.

So we have the following flow here:
```
--- continue() (first time)
--- --- Promise.resolve().then(() => onsuccess())
--- --- --- continue()
--- --- --- --- Promise.resolve().then(() => onsuccess())
--- --- --- --- --- continue()
...........
```

Looks like promise somehow keeps stack of everything inside callback even if ```Promise.resolve``` not returned from ```then``` callback.